spec/controllers/api/v1/card_decks_controller_spec.rb

```diff
@@ -91,4 +89,126 @@
       end
     end
   end
+  
+  describe 'POST #create' do
+    context 'with valid params' do
+      def call_create
+        card = FactoryBot.create(:card, user: @user)
+        deck = FactoryBot.create(:deck, user: @user)
+        @card_decks_before_request = CardDeck.count
+        @card_deck_params = FactoryBot.attributes_for(:card_deck, card_id: card.id, deck_id: deck.id)
+        
+        request.headers['X-Session-Token'] = @session.token
+        post :create, params: { card_deck: @card_deck_params }
+      end
+      
+      before :each do
+        call_create
+      end
+      
+      it 'is expected to have :created (201) HTTP response code' do
+        expect(response.status).to eq(201)
+      end
+      
+      it 'is expected to return application/json content_type' do
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+      end
+      
+      it 'is expected to call authorize (Pundit)' do
+        expect(controller).to receive(:authorize)
+        call_create
+      end
+      
+      it 'is expected to created a new CardDeck' do
+        expect(CardDeck.count).to eq(@card_decks_before_request + 1)
+      end
+      
+      it 'is expected to set fields for CardDeck' do
+        card_deck = CardDeck.order(id: :desc).first
+        expect(@card_deck_params[:card_id]).to eq(card_deck[:card_id])
```

Comment:
You should check JSON fields agains `@card_deck_params`. This test case achieves nothing.

Reply:
Sorry, actually it achieves what we need.
Why are you using `card_deck[:card_id]` syntax?